src/lib/dnssd/Resolver.h

```diff
@@ -38,7 +38,12 @@ namespace Dnssd {
 /// Node resolution data common to both operational and commissionable discovery
 struct CommonResolutionData
 {
+#ifdef CHIP_DEVICE_CONFIG_MAX_DISCOVERED_IP_ADDRESSES
     static constexpr unsigned kMaxIPAddresses = CHIP_DEVICE_CONFIG_MAX_DISCOVERED_IP_ADDRESSES;
+#else
+    static constexpr unsigned kMaxIPAddresses = 5;
```

Comment:
Usually to fix this kind of a build problem, we'd define the macro in the CHIPProjectConfig.h header files. There is a hierarchy for those files, where a constant like this can be assigned a default value at the top.

Reply:
I'll try adding it over there, it didn't seem correct since the other definitions in there are broader but i see one in there that defines `MAX_EXCHANGE_CONTENTS` so maybe this fits

Reply:
Another option, which may be easier/better, is we could just add a defines statement in the BUILD.gn file for this dnssd library: https://github.com/project-chip/connectedhomeip/blob/master/src/lib/dnssd/BUILD.gn

There are several examples of this pattern in other BUILD.gn files like this: https://github.com/project-chip/connectedhomeip/blob/master/src/platform/BUILD.gn#L98-L306

Reply:
I ended up moving this to `src/platform/BUILD.gn`. It isn't really the nicest fit in that space but since other build files pulled in `dnssd` differently this was the most top level source for it. Reworked it so that you can overwrite it with args and it defaults to 5 if unset